README.md

```diff
@@ -27,58 +26,70 @@ Repository to discuss the new REST API contract for DSpace 7. The code to implem
 
 ## Use of the HTTP Verbs and HTTP Response CODE
 
+_Please note that within this section, all terms used are meant to reference RESTful terminology and/or terminology borrowed from [Spring Data REST](https://docs.spring.io/spring-data/rest/docs/current/reference/html/#repository-resources)._ 
+ - `resource` - anything that can be identified via a URI. It's a representation of an underlying object. e.g. `/items/123-456-789` is a resource that represents a single DSpace Item. For additional examples see https://restful-api-design.readthedocs.io/en/latest/resources.html
+ - `object` - we use this term to mean the actual object *behind* the resource. In other words, a resource is a JSON representation of some object (often a DSpaceObject) stored in the database. This is also sometimes called an "entity".
+ - `collection` - a group of resources e.g. `/items` is a collection of all DSpace Item resources. This is also sometimes called a "collection resource" (e.g. in Spring Data REST)
+     - Somewhat confusingly, Spring Data REST likes to use the phrase "item resources" for individual resources within a collection. Below, we've just called them "resources".
+ - `association` - a link or relationship between two resources. This is also sometimes called an "association resource" (e.g. in Spring Data REST).
+ 
 ### On collection of resources endpoints
-- POST
-Adds a new element to the collection.
 
-- GET
-Returns the first page of the resources in the collection
+This type of endpoint interacts with a group (or collection) of resources (objects). For example: `/api/core/items` references *all* Items in the system. 
+
+- `POST`:
+Adds a new resource to the group (or collection). This creates a new object. The data for the new object _should be included_ in the request body. Related or additional information (such as required associations to other objects) may be passed as querystring parameters in the request, _if it is required to create the object._
```

Comment:
TODO for self. This needs an *example* of when we allow querystring params.  A good example is POSTing a new Item, as you cannot create an Item *without* the link (association) to the Collection.